### PR TITLE
[Improvement][Metrics] Add metrics for alert server

### LINKED ISSUE #11131: [Improvement][Metrics] Add metrics for alert server
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

* Currently we don't have any metrics in `alert server`. If `alert server` fails, with metrics, it would be easier to track down the issues. 
* Furthermore, metrics enable users to monitor alert server from external system and decrease the probability of failure.
* related: #10211

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of the pull request

* Added metrics for alert server, updated related docs and added related grafana demo dashboards.
* This PR closes: #11131 

## Brief change log

* Added metrics for alert server.

## Verify this pull request

* Verified by manual tests.

### CODE REVIEW

**Comment by EricGao888** [comment]
Coverage seems 0% for this PR, I'm not sure whether it is meaningful to find a way to cover those metrics related code.

> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=11240)
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11240&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11240&metric=new_coverage&view=list) [0.0% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11240&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11240&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11240&metric=new_duplicated_lines_density&view=list)

File: dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.function.Supplier;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Metrics;
+
+public final class AlertServerMetrics {
+
+    private AlertServerMetrics() {
+        throw new UnsupportedOperationException("Utility class");
```
Comment (kezhenxu94) [inline dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java]: Use `@UtilityClass` annotation in this class to reduce these boilerplate codes

**Review by kezhenxu94 — DISMISSED** [review]

File: dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java (hunk 1 above)
Comment (EricGao888) [reply dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java]: Sure, thx for the suggestion. Will update it.

Comment (EricGao888) [reply dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java]: Besides constructors, it looks like `@UtilityClass` automatically marks all fields and methods in the class as `static final`. Should we remove all of them during refactoring? Or keep those `static final` along with `@UtilityClass` for readability?
https://projectlombok.org/features/experimental/UtilityClass
WDYT @kezhenxu94

Comment (kezhenxu94) [reply dolphinscheduler-alert/dolphinscheduler-alert-server/src/main/java/org/apache/dolphinscheduler/alert/AlertServerMetrics.java]: I prefer to remove them. Utility class annotation takes care of all the conventions of a utility class

**Comment by EricGao888** [comment]
@kezhenxu94 @ruanwenjun Could you please take another look when available? Thanks~

**Review by kezhenxu94 — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**